packages/next/src/build/index.ts

```diff
@@ -2196,11 +2196,7 @@ export default async function build(
             exportPageWorker: sharedPool
               ? staticWorkers.exportPage.bind(staticWorkers)
               : undefined,
-            endWorker: sharedPool
-              ? async () => {
-                  await staticWorkers.end()
-                }
-              : undefined,
```

Comment:
Note that `staticWorkers.end()` doesn't take any argument so this change should be safe.